src/plugins/workspace/public/components/delete_workspace_modal/delete_workspace_modal.tsx

```diff
@@ -31,9 +50,119 @@ export function DeleteWorkspaceModal(props: DeleteWorkspaceModalProps) {
+  const targetWorkspaceId = targetWorkspaceOption?.at(0)?.key;
+  let confirmDuplicateButtonEnabled = false;
+  const [savedObjectsCount, setSavedObjectsCount] = useState(0);
+  const [allowedTypes, setAllowedTypes] = useState<string[]>([]);
+  const maxObjectsAmount: number = 200;
+  const onTargetWorkspaceChange = (targetOption: WorkspaceOption[]) => {
+    setTargetWorkspaceOption(targetOption);
+  };
+
+  if (!!targetWorkspaceId && savedObjectsCount > 0) {
+    confirmDuplicateButtonEnabled = true;
+  }
+
+  useEffect(() => {
+    const workspaceList = workspaces!.workspaceList$.value;
+    const initWorkspaceOptions = [
+      ...workspaceList
+        .filter((workspace: WorkspaceAttribute) => !workspace.libraryReadonly)
+        .filter((workspace: WorkspaceAttribute) => workspace.id !== selectedWorkspace?.id)
+        .map((workspace: WorkspaceAttribute) => workspaceToOption(workspace)),
+    ];
+    setWorkspaceOptions(initWorkspaceOptions);
+    fetchSavedObjectsCount();
+  }, [workspaces]);
+
+  const fetchSavedObjectsCount = async () => {
+    const types = await getAllowedTypes(http!);
+    const findOptions: SavedObjectsFindOptions = {
+      workspaces: [selectedWorkspace?.id as string],
+      fields: ['id'],
+      type: types,
+      perPage: 1,
+    };
+
+    try {
+      const resp = await savedObjectsClient.find(findOptions);
+      setAllowedTypes(types);
+      setSavedObjectsCount(resp.total);
+    } catch (error) {
+      notifications?.toasts.addDanger({
+        title: i18n.translate(
+          'workspace.deleteWorkspaceModal.unableFindSavedObjectsNotificationMessage',
+          { defaultMessage: 'Unable find saved objects count' }
+        ),
+        text: `${error}`,
+      });
+    }
+  };
+
+  const fetchObjects = async () => {
+    const findOptions: SavedObjectsFindOptions = {
+      workspaces: [selectedWorkspace?.id as string],
+      type: allowedTypes,
+      perPage: maxObjectsAmount,
+    };
+
+    try {
+      return await savedObjectsClient.find(findOptions);
+    } catch (error) {
+      notifications?.toasts.addDanger({
+        title: i18n.translate(
+          'workspace.deleteWorkspaceModal.unableFindSavedObjectsNotificationMessage',
+          { defaultMessage: 'Unable find saved objects' }
+        ),
+        text: `${error}`,
+      });
+    }
+  };
+
+  const moveObjectsToTargetWorkspace = async () => {
+    setIsLoading(true);
+    try {
+      for (let i = 1; i <= Math.ceil(savedObjectsCount / maxObjectsAmount); i++) {
+        const resp = await fetchObjects();
+        if (resp!.total === 0) break;
+        const objects: SaveObjectToMove[] = resp!.savedObjects.map((obj) => ({
+          id: obj.id,
+          type: obj.type,
+          workspaces: obj.workspaces!,
+          attributes: obj.attributes,
+        }));
+        await moveToTargetWorkspace(
+          http!,
+          objects,
+          selectedWorkspace?.id as string,
+          targetWorkspaceId as string
+        );
```

Comment:
What about moving all these logic to a Node.js API of workspace? There are some beneficials:
- For backend API, they can get access to all of the allowed types, so you can remove the allowed types stuff from public directory.
- The batch stuff is more like a backend concept and should be transparent to browser code.

Reply:
I will transfer the logic to backend API.